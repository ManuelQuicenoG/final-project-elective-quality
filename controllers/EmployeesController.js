import axios from 'axios';

export const employeeController = {
    register(employee) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `employees`,
            data: employee,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'employees'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    getemployee(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    updateemployee(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    }
}

