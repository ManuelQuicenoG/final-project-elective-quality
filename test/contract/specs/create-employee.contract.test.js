import { provider } from '../config/initPact';
import { employeeController } from '../../../controllers';

describe('Create an employee service', () => {
    beforeAll(async() => {
        await provider.setup();
    });
    describe('When someone wants to create an employee', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service save is ready',
                uponReceiving: 'a request to create an employee',
                withRequest: {
                    method: 'POST',
                    path: '/employees',
                    body: {'name': 'New employee',
                    'rolPosition': 'Administración',
                    'gender': 'Male',
                    'hasCourses': false,
                    'courses': []}
                },
                willRespondWith: {
                    status: 201
                }
            });
        });

        test('Then it should return correct information', async() => {
            const response = await employeeController.register({'name': 'New employee',
            'rolPosition': 'Administración',
            'gender': 'Male',
            'hasCourses': false,
            'courses': []});
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});