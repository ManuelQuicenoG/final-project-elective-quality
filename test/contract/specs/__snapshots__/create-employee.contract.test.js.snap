// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create an employee service When someone wants to create an employee Then it should return correct information 1`] = `""`;
